_source/_posts/2022-01-13-spring-cloud-contract.md

```diff
@@ -0,0 +1,893 @@
+```
+
+You're specifying the `com.example:producer` project, with the latest version, the stubs (not the jar), and telling the stub runner to start the stubbed service on port `8100`. Putting the stub runner in `LOCAL` mode means that it will look in the local Maven repo instead of looking online, which can be configured using the `REMOTE` value here. With that single annotation, Spring will create a service on port `8100` based on the stubs and generated from the contracts in the producer project. 
+
+You also define three tests that match the expected behavior defined in the contracts. 
+
+Next, create a more realistic integration test on the consumer in a new file, `ContractProducerTest.java`.
+
+`src/test/java/com/example/consumer/ContractProducerTest.java`
+
+```java
+package com.example.consumer;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.contract.stubrunner.spring.AutoConfigureStubRunner;
+import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties;
+import org.springframework.test.web.servlet.MockMvc;
+import static org.hamcrest.Matchers.containsString;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@AutoConfigureStubRunner(ids = {"com.example:producer:+:stubs:8100"}, stubsMode = StubRunnerProperties.StubsMode.LOCAL)
+public class ConsumerIntegrationTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    public void shouldReturnDefaultMessage() throws Exception {
+        this.mockMvc.perform(
+                get("/wearhat/1").header("authorization", "dummyvalue"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(content().string(containsString("Enjoy your new Test Hat 1")));
+    }
+
+}
+
+```
+
+This is why you specified the producer port in the `application.properties` file. By using different ports in `src/test/java` and `src/main/java` you are able to direct the tests to the stubbed producer, which allows you to create a true integration test. In the test above, the mocked consumer calls the stubbed producer and returns the result, which is checked against the expected value. Otherwise, when you ran the test above, it would try and call the producer on port `8080`, which would call the actual producer running on that port.
+
+// todo: add information on how to run the tests. ContractProducerTest#get_hat2() fails for me
```

Comment:
@moksamedia `ContractProducerTest#get_hat2()` fails for me. I created the apps from scratch and created a PR for your repo with my changes. https://github.com/moksamedia/okta-spring-cloud-contract/pull/1